### PR TITLE
Inefficient CMake Compile

### PR DESCRIPTION
**Pull request by psychocoderHPC**
The PIConGPU CmakeLists.txt is separating host only and accelerator source files.
In the current implementation the host only and accelarator code is compiled twice (with host and device compiler).
The result of this bug is that compiling takes up to a factor of two longer than needed and all symboles are defined twice which can lead in linker issues.
This bug was introduced with #2178.

Fix: compile `main.cpp` with the device compiler and all other files with the host compiler

### CODE REVIEW

File: include/picongpu/CMakeLists.txt [hunk 1]
```diff
@@ -384,7 +386,7 @@ target_link_libraries(picongpu-hostonly PUBLIC ${LIBS})
 
 
 cupla_add_executable(picongpu
-    ${SRCFILES}
+    "main.cpp"
```
Comment (ax3l) [inline include/picongpu/CMakeLists.txt]: `${CMAKE_CURRENT_SOURCE_DIR}/main.cpp` ?

File: include/picongpu/CMakeLists.txt [hunk 2]
```diff
@@ -375,6 +375,8 @@ include_directories(BEFORE ${PIC_EXTENSION_PATH}/include)
 ################################################################################
 
 file(GLOB_RECURSE SRCFILES "*.cpp")
```
Comment (ax3l) [inline include/picongpu/CMakeLists.txt]: maybe we just set a "DEVICESRCFILES" variable here so we have it all together (and not scattered)

File: include/picongpu/CMakeLists.txt (hunk 1 above)
Comment (psychocoderHPC) [reply include/picongpu/CMakeLists.txt]: Yes must be the absolute pass, I missed to add my last change^^

File: include/picongpu/CMakeLists.txt (hunk 2 above)
Comment (ax3l) [reply include/picongpu/CMakeLists.txt]:
```cmake
# host-only sources
file(GLOB_RECURSE SRCFILES "*.cpp")
# device sources containing kernels
set(DEVICESRCFILES "${CMAKE_CURRENT_SOURCE_DIR}/main.cpp")

# separate host only and accelerator code
list(REMOVE_ITEM SRCFILES ${DEVICESRCFILES})
```

Comment (psychocoderHPC) [reply include/picongpu/CMakeLists.txt]: note `list(REMOVE_ITEM SRCFILES ${ACCSRCFILES})` is an undocumented feature in cmake

File: include/picongpu/CMakeLists.txt [hunk 3]
```diff
@@ -374,17 +374,25 @@ include_directories(BEFORE ${PIC_EXTENSION_PATH}/include)
 # Compile & Link PIConGPU
 ################################################################################
 
+# files with accelerator code (kernel and API calls)
+list(APPEND ACCSRCFILES
+    "${CMAKE_CURRENT_SOURCE_DIR}/main.cpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/versionFormat.cpp"
+    #"${CMAKE_CURRENT_SOURCE_DIR}/plugins/common/particlePatches.cpp"
```
Comment (ax3l) [inline include/picongpu/CMakeLists.txt]: ;-) outdated

**Pull request merged**